types/basetypes/list_value_test.go

```diff
@@ -161,9 +162,9 @@ func TestNewListValueFrom(t *testing.T) {
 			expectedDiags: diag.Diagnostics{
 				diag.NewAttributeErrorDiagnostic(
 					path.Empty(),
-					"List Type Validation Error",
-					"An unexpected error was encountered trying to validate an attribute value. This is always an error in the provider. Please report the following to the provider developer:\n\n"+
-						"expected List value, received tftypes.Value with value: tftypes.String<\"oops\">",
+					"Value Conversion Error",
+					"An unexpected error was encountered trying to convert the Terraform value. This is always an error in the provider. Please report the following to the provider developer:\n\n"+
+						"can't use tftypes.String<\"oops\"> as value of List with ElementType basetypes.StringType, can only use tftypes.String values",
```

Comment:
Seems that this particular error message is being returned from the call to `(ListType).ValueFromTerraform`:

```go
	if !in.Type().Equal(l.TerraformType(ctx)) {
		return nil, fmt.Errorf("can't use %s as value of List with ElementType %T, can only use %s values", in.String(), l.ElementType(), l.ElementType().TerraformType(ctx).String())
	}
```

We could update these error messages, but it might be worth getting input from the rest of the team. Happy to open an issue for us to consider this further.